include/h2o/socketpool.h

```diff
@@ -110,7 +117,8 @@ typedef void (*h2o_socketpool_connect_cb)(h2o_socket_t *sock, const char *errstr
 /**
  * initializes a specific socket pool
  */
-void h2o_socketpool_init_specific(h2o_socketpool_t *pool, size_t capacity, h2o_url_t *origins, size_t origin_len);
+void h2o_socketpool_init_specific(h2o_socketpool_t *pool, size_t capacity, h2o_url_t *origins, size_t origin_len,
+                                  const h2o_balancer_callbacks_t *lb_callbacks, void *lb_conf, h2o_socketpool_target_conf_t *target_conf);
```

Comment:
How about making two adjustments here?

Firstly, I think that it might make sense to merge lb_callbacks and lb_conf into one argument. For example, you could create something like the following, and pass a pointer to it as an object. Each balancer logic could extend h2o_balancer_t by extending the object (by using the  `super` pattern we have in other places; e.g. `h2o_http1_conn_t`).
```
typedef struct h2o_balancer_t {
    const h2o_balancer_callbacks_t *callbacks;
};
```
Another merit of the approach is that by doing so we can get rid of the `construct` callback. Round-robin and least-conn should have their own public functions for instantiating `h2o_balancer_t`, and that function can set the `callbacks` pointer.

Secondly, I am afraid that passing two arrays that need to be handled in sync (i.e.g, `origins` and `target_conf`) might become a source of bugs. It might be worth to consider passing an array of `h2o_socketpool_target_t` as an argument (in such case, you would need to expose public functions for manipulating (e.g. create, destroy, ...) `h2o_socketpool_target_t`).